### PR TITLE
82 implement stage 2

### PR DESCRIPTION
**Pull request by Tremirre**
Implemented the second stage for our app + fixed issue with `elements` page builder property

Preview:
Preview

### CODE REVIEW

**Comment by zuzg** [comment]
Shouldn't filtering by Z threshold also happen in this stage?

**Comment by Tremirre** [comment]
> is a method `filter_low_quality_plates` for that.

Aaa you're right, forgot about it, thx

**Comment by Tremirre** [comment]
> > is a method `filter_low_quality_plates` for that.
> 
> hmm I am not sure if it shoudn't be in stage 3 actually

Unless we do not want to display invalid plates

**Comment by zuzg** [comment]
I agree, I will implement it in stage 3

File: dashboard/pages/primary_screening/callbacks.py [hunk 1]
```diff
@@ -49,6 +56,129 @@ def upload_bmg_data(contents, names, last_modified, stored_uuid, file_storage):
     )
 
 
+# === STAGE 2 ===
+
+DISPLAYED_PLATES = 9
```
Comment (zuzg) [inline dashboard/pages/primary_screening/callbacks.py:61]: visual suggestion: I would show the plates on 3x4 grid and decrease the size of the barcode font to reduce gaps

**Review by zuzg — APPROVED** [review]

File: dashboard/pages/primary_screening/callbacks.py (hunk 1 above)
Comment (Tremirre) [reply dashboard/pages/primary_screening/callbacks.py:61]: Its a good Idea but unfortunatly it significantly impacts the styling (figure and table do not want to overflow the page size and it forces page to be scrollable):

image

Comment (Tremirre) [reply dashboard/pages/primary_screening/callbacks.py:61]: + the table will have to be scrollable to fit

**Review by azywot — APPROVED** [review]

File: dashboard/pages/primary_screening/callbacks.py (hunk 1 above)
Comment (AndrzejKaj) [reply dashboard/pages/primary_screening/callbacks.py:61]: In my opinion no scrollable page and table look much better and allows for easier analysis

Comment (zuzg) [reply dashboard/pages/primary_screening/callbacks.py:61]: Okay, I see. We can leave it for now then and wait for client's feedback

**Review by AndrzejKaj — APPROVED** [review]
Everything looks very nice, visualization in my opinion is very good

**Comment by Tremirre** [comment]
I modified the styles slightly so that there is no page overflow on smaller screens (e.g. laptop ones - previously I testet it only on my PC monitor) + changed table font to something more pleasant

image

I managed to force plotly chart to be responsive (won't cause page overflow regardless of the content), yet I failed to find a reliable way of achieving simal results for a DataTable - hence I set up a few breakpoints with `@media` rule

**Comment by Tremirre** [comment]
(Horizontal Table scroll will have to stay if we want to get rid of page overflow on smaller screens - IMO a fair tradeoff)

**Pull request merged**